CMakeLists.txt

```diff
@@ -224,60 +268,43 @@ if (CUDA_FOUND AND EPIC_ENABLE_CUDA)
     cuda_add_library(epiccuda STATIC ${sources} ${headers})
     set_target_properties(epiccuda PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
 
-    set(SRC_SOLVER  src/tools/solver_manager.cpp ${GRPC_SRCS})
+    # utils, hash and grpc codes used in solver
+    set(SOLVER_UTILS_SRCS
+            src/utils/threadpool.cpp
+            src/utils/arith_uint256.cpp
+            src/utils/big_uint.cpp
+            src/utils/utilstrencodings.cpp
+            src/utils/memory/cleanse.cpp
+            )
+
+    set(SOLVER_HASH_SRCS
+            src/hash/hash.cpp
+            src/hash/sha256.cpp
+            src/hash/blake2b.cpp
+            )
+
+    set(SOLVER_RPC
+            src/rpc/basic_rpc_server.cpp)
+
+    set(SRC_SOLVER  src/tools/solver_manager.cpp ${SOLVER_RPC} ${SOLVER_UTILS_SRCS} ${SOLVER_HASH_SRCS})
     add_library(solver_lib STATIC ${SRC_SOLVER})
-    target_link_libraries(epiccore solver_lib)
-    target_link_libraries(epiccore epiccuda)
+
+    target_link_libraries(solver_lib cucakroo)
+    target_link_libraries(solver_lib epiccuda)
+    target_link_libraries(solver_lib epic_grpc)
+    target_link_libraries(solver_lib protobuf::libprotobuf)
+    target_link_libraries(solver_lib gRPC::grpc++_reflection)
 
     add_executable(solver src/tools/epic_solver.cpp)
```

Comment:
I suggest put the files epic_solver.cpp, solver_manager.cpp, solver_manager.h, solver_protocol.h and solver_task.h in a directory under src